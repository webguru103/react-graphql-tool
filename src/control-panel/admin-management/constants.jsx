import React from 'react';
import { FormattedMessage } from 'react-intl';

const DEFAULT_PAGE_SIZE = 10;

export default {
  DEFAULT_PAGE_SIZE,
  INVALID_EMAIL: <FormattedMessage id="invalid-email" defaultMessage="Please check the email format" />,
  EMPTY_EMAIL: <FormattedMessage id="empty-email" defaultMessage="Please input an email" />,
  EMPTY_FIRST_NAME: <FormattedMessage id="empty-first-name" defaultMessage="Please input a first name" />,
  EMPTY_LAST_NAME: <FormattedMessage id="empty-last-name" defaultMessage="Please input a last name" />,
  REQUEST_ERROR: <FormattedMessage id="request-error" defaultMessage="Sorry, there was an error processing your request!" />,
  SUCCESSFULLY_INVITED: <FormattedMessage id="successfully-invited" defaultMessage="The invite has been sent" />,
};
