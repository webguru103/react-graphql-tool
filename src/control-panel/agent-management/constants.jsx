import React from 'react';
import { FormattedMessage } from 'react-intl';

const DEFAULT_PAGE_SIZE = 10;

export default {
  DEFAULT_PAGE_SIZE,
  INVITES_SUCCESSFULY_SENT:
    (amount: number) => (amount === 1
      ? <FormattedMessage id="successfuly-sent-invite" defaultMessage="Invite has been sent" />
      : <FormattedMessage id="successfuly-sent-invites" defaultMessage="Successfuly sent {amount} invites" values={{ amount }} />),
};
