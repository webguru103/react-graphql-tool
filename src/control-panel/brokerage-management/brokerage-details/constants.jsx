import React from 'react';
import { FormattedMessage } from 'react-intl';

const DEFAULT_PAGE_SIZE = 10;

export default {
  DEFAULT_PAGE_SIZE,
  EMPTY_BO_NAME: <FormattedMessage id="empty-bo-name" defaultMessage="Please input the brokerage office name" />,
  EMPTY_FIELD: <FormattedMessage id="empty-field" defaultMessage="Field cannot be empty" />,
  INVALID_PHONE_NUMBER: <FormattedMessage id="invalid-phone-number" defaultMessage="Invalid Phone Number" />,
  BO_EXISTS: <FormattedMessage id="bo-exists" defaultMessage="Brokerage Office already exists" />,
  EMAIL_EXISTS: <FormattedMessage id="email-exists" defaultMessage="User with that email already exists" />,
  BO_NAME_TOO_LONG: <FormattedMessage id="bo-name-too-long" defaultMessage="Sorry, brokerage office name must be less than 64 characters" />,
  ERROR_ADDING_BO: <FormattedMessage id="error-adding-bo" defaultMessage="Sorry, there was an error adding your brokerage office!" />,
  ERROR_ADDING_ADMIN: <FormattedMessage id="error-adding-admin" defaultMessage="Sorry, there was an error adding your user!" />,
  ERROR_UPDATING_BO: <FormattedMessage id="error-updating-bo" defaultMessage="Sorry, there was an error updating your brokerage office!" />,
  SETTINGS: <FormattedMessage id="settings" defaultMessage="Settings" />,
  SET_SUPER: <FormattedMessage id="set-super" defaultMessage="Set Super Admin" />,
  NEW_BO: <FormattedMessage id="new-bo" defaultMessage="New Brokerage Office" />,
  UPDATE_BO: <FormattedMessage id="update-bo" defaultMessage="Edit Brokerage Office" />,
  CREATE: <FormattedMessage id="create" defaultMessage="Create" />,
  SUBMIT: <FormattedMessage id="submit" defaultMessage="Submit" />,
  NEXT: <FormattedMessage id="next" defaultMessage="Next" />,
};
