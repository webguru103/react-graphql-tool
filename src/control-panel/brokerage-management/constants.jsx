import React from 'react';
import { FormattedMessage } from 'react-intl';

const DEFAULT_PAGE_SIZE = 10;
const BROKERAGE_NAME_LENGTH = 100;
const UNIT_LENGTH = 10;
const STREET_NUM_LENGTH = 10;
const STREET_NAME_LENGTH = 100;
const CITY_LENGTH = 40;
const PROVINCE_LENGTH = 20;
const COUNTRY_LENGTH = 20;
const POSTAL_LENGTH = 10;

export default {
  DEFAULT_PAGE_SIZE,
  BROKERAGE_NAME_LENGTH,
  UNIT_LENGTH,
  STREET_NUM_LENGTH,
  STREET_NAME_LENGTH,
  CITY_LENGTH,
  PROVINCE_LENGTH,
  COUNTRY_LENGTH,
  POSTAL_LENGTH,
  EMPTY_BO_NAME: <FormattedMessage id="empty-bo-name" defaultMessage="Please input the brokerage office name" />,
  EMPTY_PHONE: <FormattedMessage id="empty-phone" defaultMessage="Please input a phone number" />,
  EMPTY_STREET_NUM: <FormattedMessage id="empty-street-num" defaultMessage="Please input a street number" />,
  EMPTY_STREET_NAME: <FormattedMessage id="empty-street-name" defaultMessage="Please input a street name" />,
  EMPTY_CITY: <FormattedMessage id="empty-city" defaultMessage="Please input a city" />,
  EMPTY_PROVINCE: <FormattedMessage id="empty-province" defaultMessage="Please select a province" />,
  EMPTY_POSTAL: <FormattedMessage id="empty-postal" defaultMessage="Please input a postal code" />,
  EMPTY_COUNTRY: <FormattedMessage id="empty-country" defaultMessage="Please select a country" />,
  EMPTY_FIELD: <FormattedMessage id="empty-field" defaultMessage="Field cannot be empty" />,
  INVALID_EMAIL: <FormattedMessage id="invalid-email" defaultMessage="Invalid Email" />,
  INVALID_PHONE_NUMBER: <FormattedMessage id="invalid-phone-number" defaultMessage="Invalid Phone Number" />,
  INVALID_FAX_NUMBER: <FormattedMessage id="invalid-fax-number" defaultMessage="Invalid Fax Number" />,
  INVALID_STREET_NUMBER: <FormattedMessage id="invalid-street-number" defaultMessage="Invalid Street Number" />,
  BO_EXISTS: <FormattedMessage id="bo-exists" defaultMessage="Brokerage Office already exists" />,
  EMAIL_EXISTS: <FormattedMessage id="email-exists" defaultMessage="User with that email already exists" />,
  BO_NAME_TOO_LONG: <FormattedMessage id="bo-name-too-long" defaultMessage="Sorry, brokerage office name must be less than 100 characters" />,
  UNIT_TOO_LONG: <FormattedMessage id="unit-too-long" defaultMessage="Sorry, unit must be less than 10 characters" />,
  STREET_NUM_TOO_LONG: <FormattedMessage id="street-num-too-long" defaultMessage="Sorry, street number must be less than 10 characters" />,
  STREET_NAME_TOO_LONG: <FormattedMessage id="street-name-too-long" defaultMessage="Sorry, street name must be less than 100 characters" />,
  CITY_TOO_LONG: <FormattedMessage id="city-too-long" defaultMessage="Sorry, city must be less than 40 characters" />,
  PROVINCE_TOO_LONG: <FormattedMessage id="province-too-long" defaultMessage="Sorry, province name must be less than 20 characters" />,
  COUNTRY_TOO_LONG: <FormattedMessage id="country-too-long" defaultMessage="Sorry, country name must be less than 100 characters" />,
  POSTAL_TOO_LONG: <FormattedMessage id="postal-too-long" defaultMessage="Sorry, postal code must be less than 10 characters" />,
  ERROR_ADDING_BO: <FormattedMessage id="error-adding-bo" defaultMessage="Sorry, there was an error adding your brokerage office!" />,
  ERROR_ADDING_ADMIN: <FormattedMessage id="error-adding-admin" defaultMessage="Sorry, there was an error adding your user!" />,
  ERROR_CHECKING_EMAIL: <FormattedMessage id="error-checking-email" defaultMessage="Sorry, there was an error finding that email" />,
  ERROR_GETTING_USER: <FormattedMessage id="error-getting-user" defaultMessage="Sorry, there was an error finding that user" />,
  ERROR_GETTING_BROKERAGES: <FormattedMessage id="error-gettting-brokerages" defaultMessage="Sorry, there was an error retrieving brokerages" />,
  SUCCESS_CREATE_BROKERAGE: <FormattedMessage id="success-create-brokerage" defaultMessage="Brokerage Office created" />,
  SUCCESS_INVITE_ADMIN: <FormattedMessage id="success-invite-admin" defaultMessage="Invitation sent" />,
  SETTINGS: <FormattedMessage id="settings" defaultMessage="Settings" />,
  SET_SUPER: <FormattedMessage id="set-super" defaultMessage="Set Super Admin" />,
  NEW_BO: <FormattedMessage id="new-bo" defaultMessage="New Brokerage Office" />,
  CREATE: <FormattedMessage id="create" defaultMessage="Create" />,
  INVITE: <FormattedMessage id="invite" defaultMessage="Invite" />,
  ADD: <FormattedMessage id="add" defaultMessage="Add" />,
  SUBMIT: <FormattedMessage id="submit" defaultMessage="Submit" />,
  CREATE_CONT: <FormattedMessage id="create-cont" defaultMessage="Create and Continue" />,
  NOT_NOW: <FormattedMessage id="not-now" defaultMessage="Not Right Now" />,
  WANT_TO_ADD_ADMIN: <FormattedMessage id="want-to-add-admin" defaultMessage="Adding an Admin to this BO?" />,
  ADD_ADMIN: <FormattedMessage id="add-admin" defaultMessage="Add Admin Access" />,
  INVITE_ADMIN: <FormattedMessage id="invite-admin" defaultMessage="Invite New Admin" />,
};
