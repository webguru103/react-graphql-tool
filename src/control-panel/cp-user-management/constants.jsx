import React from 'react';
import { FormattedMessage } from 'react-intl';

const DEFAULT_PAGE_SIZE = 10;

export default {
  DEFAULT_PAGE_SIZE,
  INVALID_EMAIL: <FormattedMessage id="invalid-email" defaultMessage="Please check the email format" />,
  EMPTY_EMAIL: <FormattedMessage id="empty-email" defaultMessage="Please input an email" />,
  EMPTY_FIRST_NAME: <FormattedMessage id="empty-first-name" defaultMessage="Please input a first name" />,
  EMPTY_LAST_NAME: <FormattedMessage id="empty-last-name" defaultMessage="Please input a last name" />,
  EMAIL_ALREADY_EXISTS: <FormattedMessage id="email-already-exists" defaultMessage="Email already exists as a CP User" />,
  REQUEST_ERROR: <FormattedMessage id="request-error" defaultMessage="Sorry, there was an error processing your request!" />,
  INVITE_SENT: <FormattedMessage id="invite-sent" defaultMessage="Invitation Sent!" />,
  USER_WILL_BE_REMOVED: <FormattedMessage id="user-will-be-removed" defaultMessage="This user will no longer have access to Control Panel" />,
  CP_USER_REMOVED: <FormattedMessage id="cp-user-removed" defaultMessage="CP User Removed" />,
  REMOVE_CP_USER: <FormattedMessage id="remove-cp-user" defaultMessage="Remove" />,
};
