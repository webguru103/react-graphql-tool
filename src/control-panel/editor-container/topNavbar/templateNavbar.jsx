import * as React from 'react';
import { withRouter } from 'react-router';
import { withStyles } from '@material-ui/core';
import IconButton from '@material-ui/core/IconButton';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';
import Button from '../../../shared/button/button';
import { withDialog } from '../../../shared/dialog/withDialog';
import TemplateBreadcrumb from '../templateBreadcrumb';
import { withGroupName } from '../api/editor.service';
import styles from './navbar.styles';
import { compose, get } from '../../../utility';
import messages, { EDITOR_MODE } from '../../../editor/constants';
import ConfirmPublishDialog from '../confirmPublishDialog';
import DiscardDraftDialog from '../discardDraftDialog';
import type { Form } from '../../../flowTypes';

type Props = {
  classes: Object,
  history: Object,
  createDialog: Function,
  mode: $Values<typeof EDITOR_MODE>,
  groupName: string,
  form: Form,
  transactionName: string,
  match: { params: { transactionId: string }},
  refreshForm: () => void,
};

export const TopNavbarC = ({
  classes,
  history,
  createDialog,
  mode,
  groupName,
  form,
  transactionName,
  refreshForm,
}: Props) => {

  const onArrowBack = () => {
    if (get(history, 'location.state.fromDraft', false)) {
      history.push(`/cp-user/editor/edit?formId=${form.id}`, { fromDraft: false });
      return;
    }

    history.push(`/cp-user/form-manager/${form.formGroupId}`);
  };

  return (
    <div className={classes.navBar}>
      <div className={classes.leftNav}>
        <IconButton
          classes={{ root: classes.arrowBackButton }}
          disableRipple
          onClick={onArrowBack}
        >
          <ArrowBackIcon />
        </IconButton>
        <div className={classes.breadcrumbs}>
          <TemplateBreadcrumb
            groupName={groupName}
            formName={form && form.formName}
            published={mode === EDITOR_MODE.TEMPLATE_PUBLISHED}
          />
          <div className={classes.transactionName}>
            {transactionName}
          </div>
        </div>
      </div>
      <div className={classes.rightNav}>
        {
          mode === EDITOR_MODE.TEMPLATE_DRAFT
          && form
          && form.publishedVersionId
          && (
            <Button
              secondary
              onClick={() => {
                history.push(`/cp-user/editor/view?formId=${form.id}`, { fromDraft: true });
              }}
            >
              View Published Version
            </Button>
          )
        }
        {
          form && mode === EDITOR_MODE.TEMPLATE_DRAFT && (
            <Button
              testId="discard-draft"
              secondary
              disabled={!form.publishedVersionId}
              onClick={() => createDialog({
                dialogContent: <DiscardDraftDialog
                  formId={form.id}
                  formName={form.formName}
                  groupId={form.formGroupId}
                  refreshForm={refreshForm}
                />,
              })}
            >
              {(mode === EDITOR_MODE.TEMPLATE_DRAFT) && (messages.DISCARD_DRAFT)}
            </Button>
          )
        }
        {
          form && mode === EDITOR_MODE.TEMPLATE_DRAFT && get(form, 'formVersion.id', -1) === form.draftVersionId
            && (
              <Button
                testId="publish-form"
                onClick={() => createDialog({
                  dialogContent: <ConfirmPublishDialog
                    formName={form && form.formName}
                    formId={form.id}
                  />,
                })}
              >
                {messages.PUBLISH}
              </Button>
            )
          }
        {
          mode === EDITOR_MODE.TEMPLATE_PUBLISHED && form && get(form, 'formVersion.id', -1) === form.publishedVersionId
          && (
            <Button onClick={() => history.push(`/cp-user/editor/edit?formId=${form.id}`)}>Edit Draft</Button>
          )
        }
      </div>
    </div>
  );

};

export default compose(
  withRouter,
  withGroupName,
  withStyles(styles, { withTheme: true }),
  withDialog,
)(TopNavbarC);
