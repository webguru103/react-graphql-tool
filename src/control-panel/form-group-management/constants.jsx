import React from 'react';
import { FormattedMessage } from 'react-intl';

export const FORM_GROUP_VISIBILITY = {
  EVERYONE: 'EVERYONE',
  OREA: 'OREA',
  TREB: 'TREB',
  LSTAR: 'LSTAR',
  NO_ONE: 'NO_ONE',
};

export default {
  EMPTY_GROUP_NAME: <FormattedMessage id="empty-group-name" defaultMessage="Please input a group name" />,
  EMPTY_VISIBILITY: <FormattedMessage id="empty-visibility" defaultMessage="Please input a visibility selection" />,
  GROUP_NAME_NOT_UNIQUE: <FormattedMessage id="group-name-not-unique" defaultMessage="Group Name is not unique" />,
  GROUP_CREATED_SUCCESSFULLY: <FormattedMessage id="group-created-successfully" defaultMessage="Form group created successfully" />,
  REQUEST_ERROR: <FormattedMessage id="request-error" defaultMessage="Sorry, there was an error processing your request!" />,
  CANCEL: <FormattedMessage id="cancel" defaultMessage="Cancel" />,
  CONFIRM: <FormattedMessage id="confirm" defaultMessage="Confirm" />,
  SETTINGS: <FormattedMessage id="settings" defaultMessage="Settings" />,
  FORM_GROUP_UPDATED: <FormattedMessage id="form_group_updated" defaultMessage="Succesfully Updated Form Group" />,
};
