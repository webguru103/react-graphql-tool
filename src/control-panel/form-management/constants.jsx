import React from 'react';
import { FormattedMessage } from 'react-intl';

export const FORM_STATUS = {
  ACTIVE: 'ACTIVE',
  INACTIVE: 'INACTIVE',
};

export const MAX_UPLOAD_SIZE = 25 * 1024 * 1024;
export const PDF = 'application/pdf';

export default {
  NEW_FORM: <FormattedMessage id="new-form" defaultMessage="New Form" />,
  UPLOAD_LABEL: <FormattedMessage id="upload-label" defaultMessage="Upload PDF" />,
  FORM_NAME: <FormattedMessage id="form-name" defaultMessage="Form Name" />,
  NO_FORM_BASE_UPLOADED: <FormattedMessage id="no-pdf-uploaded" defaultMessage="Please upload the form base" />,
  START_EDITING: <FormattedMessage id="start-editing" defaultMessage="Start Editing" />,
  EMPTY_FORM_NAME: <FormattedMessage id="empty-form-name" defaultMessage="Please input the form name" />,
  FORM_NAME_LENGTH: 64,
  FORM_NAME_TOO_LONG: <FormattedMessage id="form-name-too-long" defaultMessage="Sorry, name must be less than 64 characters" />,
  FORM_UPDATED: <FormattedMessage id="form-updated" defaultMessage="Form updated" />,
  NETWORK_ERROR: <FormattedMessage id="network-error" defaultMessage="Some network error occurred, try again later." />,
  NOT_EMPTY: <FormattedMessage id="form-name-empty" defaultMessage="Should not be empty" />,
  ACTIVE: <FormattedMessage id="active" defaultMessage="Active" />,
  INACTIVE: <FormattedMessage id="inactive" defaultMessage="Inactive" />,
};
