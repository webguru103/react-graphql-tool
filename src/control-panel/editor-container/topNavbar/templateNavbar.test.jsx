import React from 'react';
import {
  render, cleanup,
} from 'react-testing-library';
import { withContext } from '../../../../tests/test-helpers/contextWrapper';
import { TopNavbarC } from './templateNavbar';
import { EDITOR_MODE } from '../../../editor/constants';

const mockForm = {
  id: 1,
  formName: 'Test Form',
  draftVersionId: 2,
  formVersion: {
    id: 2,
  },
};

describe('Editor TopNavbar', () => {
  afterEach(cleanup);
  const Component = withContext(TopNavbarC);
  test('renders', () => {
    const { container } = render(
      <Component classes={{}} mode={EDITOR_MODE.TEMPLATE_DRAFT} groupId={1} formId={1} groupName="Mock Group Name" form={mockForm} />,
    );
    expect(container).toMatchSnapshot();
  });
});
