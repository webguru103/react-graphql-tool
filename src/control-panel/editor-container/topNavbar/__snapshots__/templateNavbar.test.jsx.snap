// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor TopNavbar renders 1`] = `
<div>
  <div>
    <div>
      <button
        class="MuiButtonBase-root-7 MuiIconButton-root-1"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label-6"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-10"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
        </span>
      </button>
      <div>
        <div>
          <div
            class="TemplateBreadcrumbC-groupName-19"
          >
            Mock Group Name
          </div>
          <div
            class="TemplateBreadcrumbC-formName-20"
          >
            <span>
              Test Form
            </span>
            <span />
          </div>
        </div>
        <div />
      </div>
    </div>
    <div>
      <button
        class="MuiButtonBase-root-7 MuiButtonBase-disabled-8 MuiButton-root-25 MuiButton-contained-36 MuiButton-containedSecondary-38 ButtonC-secondary-23 MuiButton-raised-39 MuiButton-raisedSecondary-41 MuiButton-disabled-45 ButtonC-disabled-24 ButtonC-button-21"
        data-testid="discard-draft"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-label-26"
        >
          <span>
            Discard Draft
          </span>
        </span>
      </button>
      <button
        class="MuiButtonBase-root-7 MuiButton-root-25 MuiButton-contained-36 MuiButton-containedPrimary-37 ButtonC-primary-22 MuiButton-raised-39 MuiButton-raisedPrimary-40 ButtonC-button-21"
        data-testid="publish-form"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-26"
        >
          <span>
            Publish
          </span>
        </span>
      </button>
    </div>
  </div>
</div>
`;
